index.js

```diff
@@ -177,12 +180,21 @@ function compression (options) {
       var negotiator = new Negotiator(req)
       var method = negotiator.encoding(['gzip', 'deflate', 'identity'], ['gzip'])
 
+      // if no method is found, use the default encoding
+      if (encodingSupported.indexOf(defaultEncoding) !== -1 && !req.headers['accept-encoding']) {
+        method = defaultEncoding === '*' ? 'gzip' : defaultEncoding
+      }
+
       // negotiation failed
       if (!method || method === 'identity') {
         nocompress('not acceptable')
         return
       }
 
+      if (opts.defaultEncoding) {
```

Comment:
Yeah, this shouldn't be necessary with the `enforceEncoding` variable or something else.